ibm/resource_ibm_private_dns_resource_record.go

```diff
@@ -355,12 +359,16 @@ func resourceIBMPrivateDNSResourceRecordRead(d *schema.ResourceData, meta interf
 }
 
 func resourceIBMPrivateDNSResourceRecordUpdate(d *schema.ResourceData, meta interface{}) error {
+	id_set := strings.Split(d.Id(), "/")
+	mk := "private_dns_resource_record_" + id_set[2]
```

Comment:
I don't think the lock should be on id_set[2] which is the resource record type ID...Its unique for each resource.

It should be also base don zone_id (id_set[1])

Reply:
I am testing whether update works with zone id or not.